scene/2d/tile_map.cpp

```diff
@@ -773,18 +773,7 @@ void TileMapLayer::_scenes_update_dirty_quadrants(SelfList<TileMapQuadrant>::Lis
 	SelfList<TileMapQuadrant> *q_list_element = r_dirty_quadrant_list.first();
 	while (q_list_element) {
 		TileMapQuadrant &q = *q_list_element->self();
-
-		// Clear the scenes if instance cache was cleared.
-		if (instantiated_scenes.is_empty()) {
-			for (const KeyValue<Vector2i, String> &E : q.scenes) {
-				Node *node = tile_map_node->get_node_or_null(E.value);
-				if (node) {
-					node->queue_free();
-				}
-			}
-		}
-
-		q.scenes.clear();
```

Comment:
`q.scenes()` should not be cleared if the cache exists, otherwise the scenes can't be removed anymore.